interfaces: add kconfig paths to system-observe

jnsgruk opened the pull request:
This PR adds three paths to the `system-observe` interface to allow read-only access to the kernel build config.

x-ref: https://forum.snapcraft.io/t/new-interface-proposal-kernel-config/31253

mvo5 reviewed (approved):
Thank you, this looks fine to me. A +1 from security is also needed of course.

alexmurray reviewed (changes requested):
Can you please replace the plain `/proc` with `@{PROC}`, ie:

```
@{PROC}/config.gz r,
/boot/config* r,
```

Also how standard is the `/lib/modules/*/build/.config` path? I can't find anything in the ubuntu archive which ships with this path (`apt-file search -x /lib/modules/.*/build/.config` comes up empty)? Is this strictly necessary?

jnsgruk commented:
> Can you please replace the plain `/proc` with `@{PROC}`, ie:
> 
> ```
> @{PROC}/config.gz r,
> /boot/config* r,
> ```
> 
> Also how standard is the `/lib/modules/*/build/.config` path? I can't find anything in the ubuntu archive which ships with this path (`apt-file search -x /lib/modules/.*/build/.config` comes up empty)? Is this strictly necessary?

I've fixed the `@{PROC}` path now, and removed the `/lib/modules/...` path for now. It's not standard in Ubuntu/Debian based systems, and is more common in RHEL and derivatives.  We can add back if required at a later date as required.

mvo5 commented:
> Can you please replace the plain `/proc` with `@{PROC}`, ie:
> 
> ```
> @{PROC}/config.gz r,
> /boot/config* r,
> ```
> 
> Also how standard is the `/lib/modules/*/build/.config` path? I can't find anything in the ubuntu archive which ships with this path (`apt-file search -x /lib/modules/.*/build/.config` comes up empty)? Is this strictly necessary?

Fwiw, I have several of these on my system:
```
 ls -al /lib/modules/*/build/.config 
-rw-r--r-- 1 root root 261694 Jul 12 10:51 /lib/modules/5.15.0-43-generic/build/.config
-rw-r--r-- 1 root root 261879 Aug  4 19:16 /lib/modules/5.15.0-46-generic/build/.config
-rw-r--r-- 1 root root 237508 Feb  6  2020 /lib/modules/5.4.0-14-generic/build/.config
```

All created/maintained outside of dpkg it seems:
```
$ dpkg -S /lib/modules/5.15.0-46-generic/build/.config
dpkg-query: no path found matching pattern /lib/modules/5.15.0-46-generic/build/.config
```
I have not looked deeper what exactly is providing this though.

Fwiw, fine to not include it (if it's not needed) - just wanted to share that it's there on my 22.04 system :)

jnsgruk commented:
Yep, so I think keeping the distinction of providing access to the running kernel config is probably a good place to draw the line for now.

alexmurray reviewed (approved):
LGTM - thanks for making the changes @jnsgruk
